src/tasks/copy-template-files.ts

```diff
@@ -286,28 +298,62 @@ ${hasCombinedArgs
   );
 };
 
+const setUpExternalExtensionFiles = async (
+  options: Options,
+  tmpDir: string,
+) => {
+  // 1. Create tmp directory to clone external extension
+  await fs.promises.mkdir(tmpDir);
+
+  const repository = options.externalExtension!.repository;
+  const branch = options.externalExtension!.branch;
+
+  // 2. Clone external extension
+  if (branch) {
+    await execa("git", ["clone", "--branch", branch, repository, tmpDir], {
+      cwd: tmpDir,
+    });
+  } else {
+    await execa("git", ["clone", repository, tmpDir], { cwd: tmpDir });
+  }
+};
+
 export async function copyTemplateFiles(
   options: Options,
   templateDir: string,
   targetDir: string
 ) {
-  copyOrLink = options.dev ? link : copy
+  copyOrLink = options.dev ? link : copy;
   const basePath = path.join(templateDir, baseDir);
+  const tmpDir = path.join(targetDir, EXTERNAL_EXTENSION_TMP_FOLDER);
 
   // 1. Copy base template to target directory
-  await copyBaseFiles(options, basePath, targetDir)
+  await copyBaseFiles(options, basePath, targetDir);
 
   // 2. Add "parent" extensions (set via config.json#extend field)
-  const expandedExtension = expandExtensions(options);
-  options.extensions = expandedExtension;
+  options.extensions = expandExtensions(options);
 
   // 3. Copy extensions folders
-  await copyExtensionsFiles(options, targetDir);
+  await Promise.all(options.extensions.map(async (extension) => {
+    const extensionPath = extensionDict[extension].path;
+    await copyExtensionsFiles(options, extensionPath, targetDir);
+  }));
 
-  // 4. Process templated files and generate output
+  // 4. Set up external extension if needed
+  if (options.externalExtension) {
+    await setUpExternalExtensionFiles(options, tmpDir);
+    await copyExtensionsFiles(options, path.join(tmpDir, "extension"), targetDir);
+  }
+
+  // 5. Process templated files and generate output
   await processTemplatedFiles(options, basePath, targetDir);
 
-  // 5. Initialize git repo to avoid husky error
+  // 6. Delete tmp directory
+  if (options.externalExtension) {
+    await fs.promises.rmdir(tmpDir, { recursive: true });
```

Comment:
36e57cb0fa4c629e2e5d5f4e78de8ca35aa64ddf

I did `fs.promises.rm` since `fs.promises.rmdir` was deprecated. Pablo hinted at this showing his console error too (I didn't get it). It seems to be working for me!

(Even if we are `awaiting` everything, I didn't want to use `fs.rmSync` since it completely blocks the JS event loop and it's good practice if we can avoid it)

Reply:
Yup yup !! https://github.com/scaffold-eth/create-eth/commit/36e57cb0fa4c629e2e5d5f4e78de8ca35aa64ddf , seem to work great for me too ! 

Actually I tried using some other fs rmdir async action  and it was still showing the doubled output and when I tried it with sync action it worked, so thought it could be only solved with sync action 

but yeah `await fs.rm` seem to works great !